nav2_costmap_2d/plugins/static_layer.cpp

```diff
@@ -100,8 +100,9 @@ void StaticLayer::onInitialize()
   has_updated_data_ = false;
 
   rclcpp::Rate r(10);
+  rclcpp::executors::SingleThreadedExecutor exec;
   while (!map_received_ && rclcpp::ok()) {
-    rclcpp::spin_some(node_);
+    exec.spin_node_once(node_->get_node_base_interface(), std::chrono::milliseconds(100));
```

Comment:
Should the 100ms be a parameter with a default of 100?

Reply:
I don't think the timeout really matters here. We are in a infinite loop here waiting for the map to be received. The timeout only affects how quickly we spin the infinite loop. Decreasing the time could shave a few milliseconds off the system startup time, but that's about it.